src/server/api/routers/tdarr.ts

```diff
@@ -0,0 +1,375 @@
+        })
+      ),
+      z.array(
+        z.object({
+          // Audio files - Codecs (Pie segments)
+          name: z.string(),
+          value: z.number(),
+        })
+      ),
+      z.array(
+        z.object({
+          // Audio files - Containers (Pie segments)
+          name: z.string(),
+          value: z.number(),
+        })
+      ),
+    ])
+  ),
+});
+
+export type TdarrPieSegment = {
+  name: string;
+  value: number;
+};
+
+export type TdarrStatistics = {
+  totalFileCount: number;
+  totalTranscodeCount: number;
+  totalHealthCheckCount: number;
+  failedTranscodeCount: number;
+  failedHealthCheckCount: number;
+  stagedTranscodeCount: number;
+  stagedHealthCheckCount: number;
+  pies: {
+    libraryName: string;
+    libraryId: string;
+    totalFiles: number;
+    totalTranscodes: number;
+    savedSpace: number;
+    totalHealthChecks: number;
+    transcodeStatus: TdarrPieSegment[];
+    healthCheckStatus: TdarrPieSegment[];
+    videoCodecs: TdarrPieSegment[];
+    videoContainers: TdarrPieSegment[];
+    videoResolutions: TdarrPieSegment[];
+    audioCodecs: TdarrPieSegment[];
+    audioContainers: TdarrPieSegment[];
+  }[];
+};
+
+const getNodesResponseSchema = z.record(
+  z.string(),
+  z.object({
+    _id: z.string(),
+    nodeName: z.string(),
+    nodePaused: z.boolean(),
+    workers: z.record(
+      z.string(),
+      z.object({
+        _id: z.string(),
+        file: z.string(),
+        fps: z.number(),
+        percentage: z.number(),
+        ETA: z.string(),
+        job: z.object({
+          type: z.string(),
+        }),
+        status: z.string(),
+        lastPluginDetails: z
+          .object({
+            number: z.string(),
+          })
+          .optional(),
+        originalfileSizeInGbytes: z.number(),
+        estSize: z.number(),
+        outputFileSizeInGbytes: z.number(),
+      })
+    ),
+  })
+);
+
+export type TdarrWorker = {
+  id: string;
+  file: string;
+  fps: number;
+  percentage: number;
+  ETA: string;
+  jobType: string;
+  status: string;
+  step: string;
+  originalSize: number;
+  estimatedSize: number;
+  outputSize: number;
+};
+
+const getStatusTableSchema = z.object({
+  array: z.array(
+    z.object({
+      _id: z.string(),
+      HealthCheck: z.string(),
+      TranscodeDecisionMaker: z.string(),
+      file: z.string(),
+      file_size: z.number(),
+      container: z.string(),
+      video_codec_name: z.string(),
+      video_resolution: z.string(),
+    })
+  ),
+  totalCount: z.number(),
+});
+
+export type TdarrQueue = {
+  array: {
+    id: string;
+    healthCheck: string;
+    transcode: string;
+    file: string;
+    fileSize: number;
+    container: string;
+    codec: string;
+    resolution: string;
+    type: 'transcode' | 'health check';
+  }[];
+  totalCount: number;
+  startIndex: number;
+  endIndex: number;
+};
+
+export const tdarrRouter = createTRPCRouter({
+  statistics: publicProcedure
+    .input(inputSchema)
```

Comment:
Simply out of consistency with other routers, don't put the schema for input in a separate element.